backend/compact-connect/stacks/persistent_stack/data_event_table.py

```diff
@@ -0,0 +1,118 @@
+    ):
+        super().__init__(
+            scope,
+            construct_id,
+            encryption=TableEncryption.CUSTOMER_MANAGED,
+            encryption_key=encryption_key,
+            billing_mode=BillingMode.PAY_PER_REQUEST,
+            removal_policy=removal_policy,
+            point_in_time_recovery=True,
+            partition_key=Attribute(name='pk', type=AttributeType.STRING),
+            sort_key=Attribute(name='sk', type=AttributeType.STRING),
+            **kwargs,
+        )
+        stack: ps.PersistentStack = ps.PersistentStack.of(self)
+
+        self.event_handler = PythonFunction(
+            self,
+            'EventHandler',
+            description='License data event handler',
+            lambda_dir='data-events',
+            index=os.path.join('handlers', 'data_events.py'),
+            handler='handle_data_events',
+            environment={'DATA_EVENT_TABLE_NAME': self.table_name, **stack.common_env_vars},
+            alarm_topic=alarm_topic,
+        )
+        self.grant_read_write_data(self.event_handler)
+        NagSuppressions.add_resource_suppressions_by_path(
+            stack,
+            f'{self.event_handler.node.path}/ServiceRole/DefaultPolicy/Resource',
+            suppressions=[
+                {
+                    'id': 'AwsSolutions-IAM5',
+                    'reason': """
+                    This policy contains wild-carded actions and resources but they are scoped to the
+                    specific actions, KMS key and Table that this lambda specifically needs access to.
+                    """,
+                },
+            ],
+        )
+        # We should specifically set an alarm for any failures of this handler, since it could otherwise go unnoticed.
+        Alarm(
+            self,
+            'EventRecieptFailureAlarm',
+            metric=self.event_handler.metric_errors(statistic=Stats.SUM),
+            evaluation_periods=1,
+            threshold=1,
+            actions_enabled=True,
+            alarm_description=f'{self.event_handler.node.path} failed to process a message batch',
+            comparison_operator=ComparisonOperator.GREATER_THAN_OR_EQUAL_TO_THRESHOLD,
+            treat_missing_data=TreatMissingData.NOT_BREACHING,
+        ).add_alarm_action(SnsAction(alarm_topic))
+
+        self.event_processor = QueuedLambdaProcessor(
+            self,
+            'DataSource',
+            process_function=self.event_handler,
+            visibility_timeout=Duration.minutes(1),
+            retention_period=Duration.hours(1),
+            max_batching_window=Duration.minutes(5),
+            max_receive_count=3,
+            batch_size=10,
+            encryption_key=encryption_key,
+            alarm_topic=alarm_topic,
+        )
+
+        Rule(
+            self,
+            'EventReceiverRule',
+            event_bus=event_bus,
+            # We will ignore ingest events, since they have sensitive data, but will store everything else
```

Comment:
Does this mean we won't get alerts on ingest event failures?

Reply:
No, it just means that we're not storing them in the events database for this feature (which will change soon, anyway).

Reply:
Ah- got it.